### PR TITLE
Fb launch json generator

### PR DESCRIPTION
**Pull request by zahra121281**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,14 +43,14 @@ This tool simplifies remote debugging by automating essential steps required to
 
 ## Usage
 To initiate remote debugging, run:
-```bash
-sudo python3 gdb_remote.py [-h] -i IP -u USERNAME -pid PROCESS_ID [-a ARCHITECTURE]
-                     [-w WORKSPCAE] [-p PORT]
+```
+bashsudo python3 gdb_remote.py [-h] -i IP -u USERNAME -pid PROCESS_ID [-a ARCHITECTURE]
```
Comment (azharivs) [inline README.md]: you need to insert a new line between bash and sudo

**Comment by azharivs** [comment]
Since the number of required python packages are increasing, please add a dependency section to your readme file with instructions on how to install the needed python modules such as ssh-utils, etc.

File: gdb_automate.py [hunk 2]
```diff
@@ -106,16 +106,27 @@ def check_libraries_in_path(core_dump_path, search_path):
...
     site_package = site.getsitepackages()[0]
     solib_path = cnf.WORKSPACE
     sysroot = cnf.WORKSPACE
     if not is_live:
         found_libs, cont = check_libraries_in_path(core_file, cnf.WORKSPACE)
+        workspace = os.path.expanduser(cnf.WORKSPACE)
+        core_file=os.path.expanduser(core_file)
+        target_path = os.path.join(workspace, os.path.basename(core_file))
+
+        if not os.path.exists(target_path):
```
Comment (azharivs) [inline gdb_automate.py]: what if you have overwritten 'core_file' with a newer version of the file but the target_path is still pointing to an older version of the core file that got copied there as a result of a previous invocation of the gdb_remote.py on the older core file? I think you should always use the fresh core file from where the user is pointing to it.

File: gdb_automate.py [hunk 3]
```diff
@@ -168,20 +182,44 @@ def run_gdb_local(app, ip, port, pid, user, pwd,
...
+                process_id=pid
+            )
+        else: 
+            generate_debug_config(
+                mode="coredump",
+                output_path="{}.vscode/launch.json".format(cnf.WORKSPACE),
+                core_path=core_file,
+                binary_path=binary_path,
+                workspace=cnf.WORKSPACE,
+                gdb_script=tmp_file.name
```
Comment (azharivs) [inline gdb_automate.py]: why not strip the \n here?

File: gdb_coredump.py [hunk 4]
```diff
@@ -77,14 +77,18 @@ def main(argv):
         'variants as listed. Examples: powerpc:common, mips:isa32r5, sparc:v9.'
     ))
     parser.add_argument('-c', '--coredump', type=str, required=True,
-                        help='Path for coredump file.')
+                        help='Direct path for coredump file.')
```
Comment (azharivs) [inline gdb_coredump.py]: what do you mean by a "Direct path"? Absolute? Relative?

File: gdb_remote.py [hunk 5]
```diff
@@ -88,7 +88,10 @@ def main(argv):
                         help='Directory where you should put all the shared-binaries/app-binary and source codes.')
     parser.add_argument('-p', '--port', type=int,
                         help='Port to set the DEFAULT_PORT. If not specified, gdb use port: 1234.')
-
+    parser.add_argument('-ui', '--user_interface', type=str, required=True,
```
Comment (azharivs) [inline gdb_remote.py]: Better if you set a default 'gdb' for this?

**Review by azharivs — CHANGES_REQUESTED** [review]
Please find my comments, thank you!

File: gdb_coredump.py (hunk 4 above)
Comment (zahra121281) [reply gdb_coredump.py]: Absolute , edited

**Pull request merged**